drivers/ethernet/dwc_xgmac/eth_dwc_xgmac.c

```diff
@@ -0,0 +1,1943 @@
+	reg_val &= CORE_MAC_TX_CONFIGURATION_SS_CLR_MSK;
+
+	switch (link_speed) {
+	case LINK_10MBIT:
+		reg_val |= CORE_MAC_TX_CONFIGURATION_SS_SET(CORE_MAC_TX_CONFIGURATION_SS_10MHZ);
+		LOG_DBG("%s: MAC link speed updated to 10Mbps", dev->name);
+		break;
+	case LINK_100MBIT:
+		reg_val |= CORE_MAC_TX_CONFIGURATION_SS_SET(CORE_MAC_TX_CONFIGURATION_SS_100MHZ);
+		LOG_DBG("%s: MAC link speed updated to 100Mbps", dev->name);
+		break;
+	case LINK_1GBIT:
+		reg_val |= CORE_MAC_TX_CONFIGURATION_SS_SET(CORE_MAC_TX_CONFIGURATION_SS_1000MHZ);
+		LOG_DBG("%s: MAC link speed updated to 1Gbps", dev->name);
+		break;
+	default:
+		LOG_ERR("%s: Invalid link speed configuration value", dev->name);
+	}
+
+	sys_write32(reg_val, ioaddr + CORE_MAC_TX_CONFIGURATION_OFST);
+}
+
+static void dwxgmac_mac_init(const struct device *dev,
+			     const struct eth_dwc_xgmac_config *const config,
+			     struct eth_dwc_xgmac_dev_data *const data)
+{
+	struct xgmac_mac_config *const mac_cfg = (struct xgmac_mac_config *)&config->mac_cfg;
+	uint32_t ioaddr = get_reg_base_addr(dev);
+	uint32_t reg_val;
+
+	/* Enable MAC HASH & MAC Perfect filtering */
+	reg_val =
+#ifndef CONFIG_ETH_DWC_XGMAC_HW_FILTERING
+		CORE_MAC_PACKET_FILTER_RA_SET(SET_BIT) | CORE_MAC_PACKET_FILTER_PM_SET(SET_BIT);
+#else
+#ifdef CONFIG_ETH__DWC_XGMAC_HW_L3_L4_FILTERING
+		CORE_MAC_PACKET_FILTER_IPFE_SET(SET_BIT) |
+#endif
+		CORE_MAC_PACKET_FILTER_HPF_SET(SET_BIT) | CORE_MAC_PACKET_FILTER_HMC_SET(SET_BIT) |
+		CORE_MAC_PACKET_FILTER_HUC_SET(SET_BIT);
+#endif
+
+	sys_write32(reg_val, ioaddr + CORE_MAC_PACKET_FILTER_OFST);
+
+	reg_val = 0;
+	for (uint32_t q = 0; q < config->num_rx_Qs; q++) {
+		reg_val |= (XGMAC_RXQxEN_DCB << (q * XGMAC_RXQxEN_SIZE_BITS));
+	}
+	sys_write32(reg_val, ioaddr + CORE_MAC_RXQ_CTRL0_OFST);
+
+	reg_val = CORE_MAC_TX_CONFIGURATION_JD_SET(SET_BIT);
+	sys_write32(reg_val, ioaddr + CORE_MAC_TX_CONFIGURATION_OFST);
+
+	reg_val = CORE_MAC_RX_CONFIGURATION_GPSLCE_SET(SET_BIT) |
+#ifdef CONFIG_ETH_DWC_XGMAC_RX_CS_OFFLOAD
+		  CORE_MAC_RX_CONFIGURATION_IPC_SET(SET_BIT) |
+#endif
+		  CORE_MAC_RX_CONFIGURATION_WD_SET(SET_BIT) |
+		  CORE_MAC_RX_CONFIGURATION_JE_SET(mac_cfg->je) |
+		  CORE_MAC_RX_CONFIGURATION_ARPEN_SET(mac_cfg->arp_offload_en) |
+		  CORE_MAC_RX_CONFIGURATION_GPSL_SET(mac_cfg->gpsl);
+
+	sys_write32(reg_val, ioaddr + CORE_MAC_RX_CONFIGURATION_OFST);
+
+	/* Configure MAC link speed */
+	eth_dwc_xgmac_update_link_speed(dev, data->link_speed);
+}
+
+static inline void dwxgmac_irq_init(const struct device *dev)
+{
+	struct eth_dwc_xgmac_dev_data *const data = (struct eth_dwc_xgmac_dev_data *)dev->data;
+	mem_addr_t reg_addr;
+	uint32_t reg_val;
+	mem_addr_t ioaddr = get_reg_base_addr(dev);
+
+	reg_addr = (ioaddr + XGMAC_DMA_BASE_ADDR_OFFSET + DMA_MODE_OFST);
+	reg_val = (sys_read32(reg_addr) & DMA_MODE_INTM_CLR_MSK);
+	sys_write32(reg_val, reg_addr);
+	data->irq_cntxt_data.dev = dev;
+}
+
+static inline void add_frags_to_pkt(struct net_pkt *rx_pkt, struct net_buf *frag1,
```

Comment:
Can you change the var names? You are adding net_buf so buffer to the packet.  buf1/buf2 would do

Reply:
Updated